frappe/public/js/frappe/form/controls/int.js

```diff
@@ -20,11 +20,10 @@ frappe.ui.form.ControlInt = frappe.ui.form.ControlData.extend({
 			});
 	},
 	eval_expression: function(value) {
-		if (typeof value==='string'
-			&& value.match(/^[0-9+-/* ]+$/)
-			// strings with commas are evaluated incorrectly
-			// for e.g 47,186.00 -> 186
-			&& !value.includes(',')) {
+		// Removes seperators of any number format
+		value = strip_number_groups(value, this.get_number_format());
```

Comment:
`get_number_format` method does not exist in `int.js`. Also, currency has it's own `eval_expression`, it should be commonified.